index.js

```diff
@@ -1,100 +1,253 @@
   const response = {
     statusCode: 200,
-    body: JSON.stringify(canvas.toDataURL()),
+    body: JSON.stringify(canvas.toDataURL())
   };
-  
+
   return response;
 }
 
-async function loadImages(objs, data) {
-  let images = [];
-  await Promise.all(objs.map((obj,index) =>
-    loadImage(`data:image/png;base64,${obj}`).then((imageObj) => {
-      paintImages(imageObj, index, data);
+async function loadItems(objs, ctx) {
+  await Promise.all(objs.map((obj, index) =>
+    loadImage(`data:image/png;base64,${obj.img}`).then((imageObj) => {
+      paintIcons(ctx, imageObj, index, obj.count);
     })
   ));
-  return images;
 }
 
-function paintImages(image, index, data){
-    const offset = 40 * index;
-    ctx.drawImage(image, x + offset, 120);
-    ctx.font = "12px Runescape";
-    ctx.textAlign = "right";
-    const count = data.loot[index].count;
-    if (count >= 1000000) {
-    count = `${Math.floor(count / 1000000)}m`;
-    }
-    if (count > 1000) {
+async function openSkills(objs) {
+  let skills = [];
+  await Promise.all(objs.map((obj) => 
+    getSkill(`${__dirname}/icons/${obj.skill}.png`).then(response => {
+      skills.push({'img': `data:image/png;base64,${response.toString('base64')}`, 'xp': obj.xp});
+    })
+  ));
+  return skills;
+}
+
+const getSkill = async(file)=>{
+  const result = await fs.readFile(file)
+  return result;
+}
+
+async function loadSkills(objs, ctx, offset) {
+  await Promise.all(objs.map((obj, index) =>
+    loadImage(obj.img).then((imageObj) => {
+      paintSkills(ctx, imageObj, index, offset, `${obj.xp} xp`);
+    })
+  ));
+}
+
+function paintSkills(ctx, image, index, offset, xp){
+  const yOffset = offset + 40 + (50 * Math.floor(index / 5));
+  const xOffset = 15 + (60 * (index % 5));
+  ctx.drawImage(image, xOffset, yOffset, 25, 25);
+  ctx.font = "16px Runescape";
+  ctx.textAlign = "center";
+  ctx.fillStyle = "#ffff00";
+  const xpInt = parseInt(xp.replace(/,/g, ''));
+  let xpGained = '';
+  if (xpInt >= 1000000) {
+    xpGained = `${Math.floor(xpInt*10 / 1000000)/10}m xp`;
+  }
+  else if (xpInt > 1000) {
+    xpGained = `${Math.trunc((xpInt * 10) / 1000 / 10)}k xp`;
+  } else {
+    xpGained = `${xpInt} xp`;
+  }
+  ctx.strokeText(xpGained, xOffset+15, yOffset+35);
+  ctx.fillText(xpGained, xOffset+15, yOffset+35);
+  // const start = Math.floor(Math.random() * 99);
+  // const finish = Math.floor(Math.random() * (99 - start) + start); 
+  // ctx.font = "14px Runescape";
+  // ctx.strokeText(`${start}-${finish}`, xOffset+15, yOffset+50);
+  // ctx.fillText(`${start}-${finish}`, xOffset+15, yOffset+50);
+}
+
+
+function paintIcons(ctx, image, index, count){
+  const xOffset = 15 + (40 * (index % 7));
+  const row = Math.floor(index/7);
+  const yOffset = 115 + row*35;
+  ctx.drawImage(image, xOffset, yOffset);
+  ctx.font = "14px Runescape";
+  ctx.textAlign = "right";
+  if (count >= 1000000) {
+    count = `${Math.floor(count *10 / 1000000)/ 10}m`;
+  }
+  else if (count > 100000) {
```

Comment:
should i be worried that 174 and 199 have different conditions but both result in `k`s?